R/math_test_func.R

```diff
@@ -0,0 +1,118 @@
+
+# Modified measure class, now responsible for evaluating the mathematical
+# test function
+MeasureMathTestFun = R6Class("MeasureMathTestfun",
+  inherit = MeasureRegr,
+  public = list(
+    test_fun = NULL,
+
+    initialize = function(test_fun) {
+      self$test_fun = test_fun
+
+      # init superclass
+      super$initialize(
+        id = "func_value",
+        range = c(-Inf, Inf),
+        minimize = TRUE
+      )
+    },
+
+    # evaluate test function (ignore "prediction" argument) based on
+    # hyperparameter values
+    score_internal = function(prediction, learner, ...) {
+      v = learner$param_set$values
+      v$budget = NULL
+      globalOptTests::goTest(unlist(v), self$test_fun)
+    }
+  )
+)
+
+# Modified learner class, now responsible for defining the domain of the math
+# function
+# tfun: character name of valid mathematical function
+# budget is currently unused
+# TODO: budget reduces noise
+LearnerMathTestFun = R6Class(
+  "LearnerMathTestFun",
+  inherit = LearnerRegr,
+
+  public = list(
+    initialize = function(test_fun) {
+      # get the upper and lower boundaries of each dimension of the test func
+      b = globalOptTests::getDefaultBounds(test_fun)
+      # get the domain dimensions of the test function
+      k = globalOptTests::getProblemDimen(test_fun)
+      # create a new parameter set based on the received func properties
+      ps = Map(
+        function(i) ParamDbl$new(paste0("x", i), lower = b$lower[i], upper = b$upper[i]),
+        seq_len(k)
+      )
+      ps = ParamSet$new(ps)
+      # add budget parameter; TODO: do something with it
+      ps$add(ParamDbl$new("budget", lower = 0, upper = 100))
```

Comment:
should this be upper = 1? `length(task$row_roles$use) / length(task$row_ids)` is bounded above by 1 I'm pretty sure (because `mlr3` does not allow repeated `row_ids`)